### ISSUE TITLE
Allow some realty employees to rent houses

### ISSUE THREAD

**Issue by taylorwiffed** [InZidiuZ/op-framework-issue-tracker#2357]
**Assurance:**
Have you read through the rules from the `README.md` file in the root folder of this GitHub repository?

Yes

**Summary:**
Write a summary of what your suggestion is about.

Allow realtors to have employees that can actually rent properties to people. 

**Reason:**
Why would your suggestion be a good change to the framework?

Right now, a lot of the realtors are available around the same time or because of liberty are not able to fly in as much as they would like. It would be really helpful if there could be tiers of employees for realty companies, and they allow a tier to actually rent houses. 

For example, my character is employed by King Estates. Curtis has a lot going on and it would be SUPER helpful if I could actually rent the houses for him instead of asking him to fly in every single time. Would allow more real estate to be rented during hours of the day when there are few if any realtors around.

**Comment by AdeptCL** [comment]
I completely agree. There are realtors in this city who have been around for many years, but they tend to miss out on people that are within the opposite time zone, because they're unable to be around during peak playing hours.

There should be a role attached to the real estate system in the /companies of **REALTOR**. Granted the employer can set their wage, but it should enable them the server permissions to rent out properties through the /companies, like a real estate owner can currently do.

**Comment by vol5** [comment]
This is a good idea. I feel it would be best to attach the "employee" server permissions to role names.

For example, if an owner gives someone the role name of "Manager". They would then have access to that company's (My Company) comp data. Enabling them to see 2 sections of info as a Manager. "Blocks" Which they can view and "Company Property" which they can click (Manage Properties) and send property offers to clients.

This gives what I feel would be "Manager Info" and is not an overreach in permissions. It gives access to Realtor Employees so they can send offers and see what block info the company has.

**Comment by sleighwhite** [comment]
Just adding on here since I'm participating in a sort of "workaround" for this at the minute. Disregard if this insight isn't helpful!

My Job is an attorney, but in-between representing people in the cells and all, I work for PDG Realty as an Assistant. I have access to Phil's personal spreadsheet with vacancies and pricing information. When people request a Realtor, I can show them properties. I have the perms to "enter" houses that are vacant and owned by PDG. But when it comes to actually selling the house, I have to "reserve" the through our own spreadsheet system. And then I just try to call/text Phil IC and if he's unavailable, I "email" him on Discord to see if he can come around. 

I'm really grateful to have to the opportunity to "enter" houses, but it would be helpful if I could also send agreements over to individuals. I'm sure that it's a very complex system, so I understand if that's not possible. Just wanted to add some additional context from my perspective. 

Thanks!

**Comment by CloudCait** [comment]
Would love this to become a thing, I've been waiting for someone to pick up a house for almost a month now but everytime I fly out they fly in as they're opposite time zones to me. I'm finding it very hard to sell houses as I'm only available a couple hours a day because of work and sleep meaning I'm just having to dig into my own money to pay off blocks because I'm missing out on people who could want a home that I have. Also, my employees have also told me that anytime they mention that the client will have to wait for me to be around to get keys, the client decides they don't want to wait and goes to a different realtor who's available there and then.

**Comment by Brundoggy** [comment]
Bumping this

**Comment by ayushpatel1007** [comment]
What i actually do now is, i have hired few agents. Whenever they are in city, they find customers, show them house, if customer liked it then they get house booked.

Now what agent do is take details of customer and send to realtors (owner of company).
The owner change status of particular house in excel sheet from avaiable to rerserved/booked so that no other agent then show that same house to another customer.
Agent share details of customer to company owner. Whenever company owner come in city/server, he continues call the customer till he don't answer call and when customer pick, they sell that house.
 
Its lots of lots of headache and work and documentation but this is temperorary idea for company owner that want agents to atleast find customer for them.

https://i.imgur.com/smFUX5V.png

**Comment by xplrerin** [comment]
I think this needs to be boosted, Allowing employees of real estate companies to send the offer will not only increase interaction, but allow companies to hire more people and open a further branch of jobs for civilians. This doesn’t need to be complex, the money can still go into the company balance and the owner of the company can reimburse the sales person like I personally have been doing for years. It would just be nice for people to have the capability to send the offers, especially when the queue is long etc.

**Comment by vol5** [comment]
Examining the recent alterations to the script, it's evident that one adjustment stands out as particularly beneficial: enhancing the accessibility of homes across the board. This modification has been widely requested by realtors and is perceived as pivotal in optimizing server performance. How Erin described above sounds great. Also, Taylors proposal, as mentioned earlier, would significantly enhance quality of life, streamlining realty transactions and providing players with greater opportunities to establish themselves promptly, thus reducing wait times.

**Comment by coalaura** [comment]
As mentioned here #3246 there are now permission levels when promoting/hiring employees.

### Employee
*Can see all company infos but not actually do anything except use /enter and /storage. (Basically like previous employees and therefore the default)*

### Realtor
*Like the Employee level but the realtor can also sell, evict and transfer properties on their own.*

### Owner / Manager
*Has permissions to do anything, except promote/hire/fire other managers. (Can hire realtors and employees tho ofc)*

![yes](https://i.shrt.day/JP79UB5CY2.webp)
